app/mailers/application_mailer.rb

```diff
@@ -36,4 +37,27 @@ def generate_unsubscribe_token(id, email_type)
   def use_custom_host
     ActionMailer::Base.default_url_options[:host] = Settings::General.app_domain
   end
+
+  def perform_deliveries?
+    return unless Rails.env.production?
+
+    ActionMailer::Base.perform_deliveries = Settings::General.smtp_settings["password"].present? ||
+      ENV["SENDGRID_API_KEY"].present?
+    ActionMailer::Base.perform_deliveries
+  end
+
+  def delivery_method
```

Comment:
mmm I'm not sure we're supposed to shadow the `delivery_method` attribute with a method. According to https://api.rubyonrails.org/classes/ActionMailer/Base.html

`delivery_method` should either be a symbol or a custom delivery object:

> Defines a delivery method. Possible values are :smtp (default), :sendmail, :test, and :file. Or you may provide a custom delivery method object e.g. MyOwnDeliveryMethodClass. See the Mail gem documentation on the interface you need to implement for a custom delivery agent.

see what returns on `main`:

```ruby
[1] pry(main)> ApplicationMailer.new.delivery_method
=> :smtp
```

Shouldn't this hash be `self.smtp_settings = hash` ?

See also https://guides.rubyonrails.org/action_mailer_basics.html#action-mailer-configuration

Reply:
Ah, that was unintentional. I originally did not plan to add application_mailer changes into this PR but thought of a quick and dirty solution before ending my day. I should've drafted this PR.

I'm changing the method name but are you also suggesting that I set smtp_settings instead of changing delivery_method_options in default?

Reply:
Updated! I went with `before_action` because, well, we don't really need `after_action` yet at this point.